addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js

```diff
@@ -0,0 +1,446 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const url = require('url');
+const yaml = require('js-yaml');
+
+const StepBase = require('@aws-ee/base-workflow-core/lib/workflow/helpers/step-base');
+const {
+  getServiceCatalogClient,
+} = require('@aws-ee/environment-type-mgmt-services/lib/environment-type/helpers/env-type-service-catalog-helper');
+const environmentStatusEnum = require('../../helpers/environment-status-enum');
+const jsYamlCustomSchema = require('../../helpers/js-yaml-custom-schema');
+
+const inPayloadKeys = {
+  requestContext: 'requestContext',
+  envId: 'envId',
+  envName: 'envName',
+  xAccEnvMgmtRoleArn: 'xAccEnvMgmtRoleArn',
+  externalId: 'externalId',
+  provisionedProductId: 'provisionedProductId',
+  existingEnvironmentStatus: 'existingEnvironmentStatus',
+};
+
+const settingKeys = {
+  envMgmtRoleArn: 'envMgmtRoleArn',
+};
+
+const pluginConstants = {
+  extensionPoint: 'env-provisioning',
+};
+
+const failureStatuses = [
+  'CREATE_FAILED',
+  'ROLLBACK_FAILED',
+  'DELETE_FAILED',
+  'UPDATE_ROLLBACK_FAILED',
+  'ROLLBACK_COMPLETE',
+  'UPDATE_ROLLBACK_COMPLETE',
+];
+const successStatuses = ['CREATE_COMPLETE', 'DELETE_COMPLETE', 'UPDATE_COMPLETE'];
+
+/**
+ * Workflow step that terminates the launch dependencies of a product created while lauching the product
+ * Dependency check is performed based on the output of the product CFT
+ */
+class TerminateLaunchDependency extends StepBase {
```

Comment:
I am guessing unit tests are pending for this class?

Reply:
Yes. You can expect unit test cases in the subsequent PRs